File: calc/calculate.cpp

#include <iostream>
#include <vector>

class Matrix {
    public:
        Matrix(const int &crows, const int &ccolumns);

        void print();
        std::vector<std::vector<int>> summation(Matrix matr2);
        int getItem(const int &i, const int &j) { return matrix[i][j]; };
        void putItem(const int &i, const int &j, int item) { matrix[i][j] = item; };    

    private:
        int rows;
        int columns;
        std::vector<std::vector<int>> matrix;
        
        void initialize(const int &crows, const int &ccolumns);
};


Matrix::Matrix(const int &crows, const int &ccolumns) {
    rows = crows;
    columns = ccolumns;
    initialize(rows, columns);
}    

void Matrix::initialize(const int &crows, const int &ccolumns) {
        matrix.resize(crows);
        for (int i = 0; i < crows; i++) {
            matrix.at(i).resize(ccolumns);
        }
}

void Matrix::print() {
    std::cout << "-----------MATRIX-----------" << std::endl;
    for (int i = 0; i < rows; i++) {
        std::cout << "         [ ";
        for (int j = 0; j < columns; j++) {
            std::cout << matrix[i][j] << ' ';
        }
        std::cout << ']' << std::endl;
    }
    std::cout << "----------------------------";
}

std::vector<std::vector<int>> Matrix::summation(Matrix matr2) {
    std::vector<std::vector<int>> resultMatrix;
    resultMatrix.resize(rows);
    for (int i = 0; i < rows; i++) {
        resultMatrix.at(i).resize(columns);
    }
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < columns; j++) {
            resultMatrix[i][j] = matr2.getItem(i, j) + matrix[i][j];
        }
    }
    return resultMatrix;
}

int main() {
    Matrix a = Matrix(3, 3);
    Matrix b = Matrix(3, 3);
    b.putItem(1, 1, 5);
    

    std::vector<std::vector<int>> c = a.summation(b);
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            std::cout << c[i][j];
        }
    }
    return 0;
}

File: calc/calculate.py
import numpy as np


class Matrix:
    def __init__(self, data: dict):
        self.nums = data["nums"]
        self.rows = data["rows"]
        self.cols = data["cols"]

        def matrInit():
            print(self.nums)
            matr = []
            i = 0
            for row in range(self.rows):
                line = []
                for col in range(self.cols):
                    line.append(self.nums[i])
                    i += 1
                matr.append(line)
            return np.array(matr)

        self.matrix = matrInit()

    def summation(self, matr2):
        data = {}
        try:
            data = {
                "matr": np.add(self.matrix, matr2.matrix),
                "rows": self.rows,
                "cols": self.cols,
                "calc": {
                    "code": 200,
                },
            }
        except IndexError:
            data["calc"] = {
                "code": 400,
                "mess": "Invalid size!",
            }
        except ValueError:
            data["calc"] = {
                "code": 400,
                "mess": "Invalid size!",
            }
        return data

    def multiply(self, item):
        data = {}
        if type(item) is not int:
            try:
                data = {
                    "matr": np.dot(self.matrix, item.matrix),
                    "rows": self.rows,
                    "cols": item.cols,
                    "calc": {
                        "code": 200,
                    },
                }
            except ValueError:
                data["calc"] = {
                    "code": 400,
                    "mess": "Invalid size!",
                }
        else:
            try:
                data = {
                    "matr": np.dot(item, self.matrix),
                    "rows": self.rows,
                    "cols": self.cols,
                    "calc": {
                        "code": 200,
                    },
                }
            except ValueError:
                data["calc"] = {
                    "code": 400,
                    "mess": "Invalid size!",
                }

        return data

    def transpose(self):
        data = {
            "matr": self.matrix.transpose(),
            "rows": self.cols,
            "cols": self.rows,
            "calc": {
                "code": 200,
            },
        }
        return data

    def findDet(self):
        data = {}
        try:
            data = {
                "matr": self.matrix,
                "rows": self.rows,
                "cols": self.cols,
                "calc": {
                    "code": 200,
                },
                "det": np.linalg.det(self.matrix),
            }
        except np.linalg.LinAlgError:
            data["calc"] = {
                "code": 406,
                "mess": "Invalid size! Det could be find only in square matrix!",
            }
        return data

    def inverse(self):
        data = {}
        try:
            data = {
                "matr": np.linalg.inv(self.matrix),
                "rows": self.rows,
                "cols": self.cols,
                "calc": {
                    "code": 200,
                },
            }
        except:
            data["calc"] = {
                "code": 406,
                "mess": "We could not find inverse matrix!",
            }
        return data


#
# pr#int(a.)
